File: src/main/java/com/github/svegon/capi/event/render/tooltip/BrewingStandTooltipCallback.java
package com.github.svegon.capi.event.render.tooltip;

import net.fabricmc.fabric.api.event.Event;
import net.fabricmc.fabric.api.event.EventFactory;
import net.minecraft.block.BrewingStandBlock;
import net.minecraft.client.item.TooltipContext;
import net.minecraft.item.ItemStack;
import net.minecraft.text.Text;
import net.minecraft.world.BlockView;
import org.jetbrains.annotations.Nullable;

import java.util.List;

@FunctionalInterface
public interface BrewingStandTooltipCallback {
    Event<BrewingStandTooltipCallback> EVENT = EventFactory.createArrayBacked(BrewingStandTooltipCallback.class,
            (block,stack, world, tooltip, options) -> {}, listeners -> (block,stack, world, tooltip, context) -> {
        for (BrewingStandTooltipCallback listener : listeners) {
            listener.appendBrewingStandTooltip(block, stack, world, tooltip, context);
        }
    });

    void appendBrewingStandTooltip(BrewingStandBlock block, ItemStack stack, @Nullable BlockView world,
                                   List<Text> tooltip, TooltipContext options);
}

File: src/main/kotlin/io/github/svegon/mclientapi/event/render/tooltip/BrewingStandTooltipCallback.kt
package io.github.svegon.capi.event.render.tooltip

import net.fabricmc.fabric.api.event.Event
import net.fabricmc.fabric.api.event.EventFactory
import net.minecraft.client.item.TooltipContext
import net.minecraft.text.Text
import net.minecraft.world.BlockView

fun interface BrewingStandTooltipCallback {
    fun appendBrewingStandTooltip(
        block: BrewingStandBlock?, stack: ItemStack?, world: BlockView?,
        tooltip: List<Text?>?, options: TooltipContext?
    )

    companion object {
        @JvmField
        val EVENT: Event<BrewingStandTooltipCallback> = EventFactory.createArrayBacked<BrewingStandTooltipCallback>(
            BrewingStandTooltipCallback::class.java,
            BrewingStandTooltipCallback { block: BrewingStandBlock?, stack: ItemStack?, world: BlockView?, tooltip: List<Text?>?, options: TooltipContext? -> }) { listeners: Array<BrewingStandTooltipCallback> ->
            BrewingStandTooltipCallback { block: BrewingStandBlock?, stack: ItemStack?, world: BlockView?, tooltip: List<Text?>?, context: TooltipContext? ->
                for (listener in listeners) {
                    listener.appendBrewingStandTooltip(block, stack, world, tooltip, context)
                }
            }
        }
    }
}
